src/main/java/de/tum/in/www1/artemis/service/connectors/ci/ContinuousIntegrationService.java

```diff
@@ -230,6 +231,7 @@ public String forProgrammingLanguage(ProgrammingLanguage language) {
                 return switch (language) {
                     case JAVA, PYTHON, HASKELL, KOTLIN, SWIFT, EMPTY -> "";
                     case C, VHDL, ASSEMBLER, OCAML -> "tests";
+                    default -> throw new IllegalArgumentException("Checkout path not defined for " + language);
```

Comment:
we want to avoid default in switch expressions, because the compiler would not warn us if new enum values are added

Reply:
Corrected in [103f8fa](https://github.com/ls1intum/Artemis/pull/9074/commits/103f8fa2b46e4efb8a960aae6db1dfdc1e8809e6).